client/protocols/analytics.py

```diff
@@ -8,9 +10,25 @@
 # TODO(albert): rename this InformationProtocol
 # Add all command line arguments here
 
+log = logging.getLogger(__name__)
+
 class AnalyticsProtocol(models.Protocol):
     """A Protocol that analyzes how much students are using the autograder."""
 
+    RE_SNIPPET = re.compile(r"""
+        \s*[\#\;]\s+BEGIN\s+(.*?)\n # \1 is question name
+        (.*)                        # \2 is the contents in between
```

Comment:
This needs to be `(.*?)` non-greedy search, or else it will grab everything until it reaches the last `END` tag:

```
BEGIN "Question 0"
stuff
END "Question 0"

BEGIN "Question 0"
boo
END "Question 0"
```

This will go all the way to the last `END`.